Task/№6 personal task

LarionovKirill opened the pull request.

LarionovKirill committed:
Начало работы. Создание проекта.

LarionovKirill committed:
Feature(AirTravel.cs): Добавлены поля класса их свойства и конструкторы класса.
Feature (FlightType.cs): Создано перечисление типов полета.
Feaure (CollisionManager.cs): Создан класс проверки коллизий.

LarionovKirill committed:
feature(UserInterface.cs): Добавил пользовательский интерфейс и сверстал часть.

LarionovKirill committed:
feature(UserInterface.cs): добавлены методы и picturebox, но это еще не все

LarionovKirill committed:
Fix(UserInterface.cs): Исправлены методы обработки полей.

LarionovKirill committed:
Feature(UserInterface.cs): Добавлена проверка на длину полета и сделаны мелкие правки.

LarionovKirill committed:
Feature(UserInterface.cs): Добавил фото, но с методами доступа к ним пока еще туго.

LarionovKirill committed:
Feature(UserINterface.cs):Добавил изменение контраста кнопок при наведении на них.

LarionovKirill committed:
Feature(UserInterface.cs): Метод добавления теперь записывает значения в ListBox

LarionovKirill committed:
Feature(UserInterface.cs): Добавлен метод удаления

LarionovKirill committed:
Feature(MainForm.cs): Изменил название формы.

LarionovKirill committed:
Feature(UserInterface.cs): добавлены методы изменения значений перелета и метод сортировки данных.

LarionovKirill committed:
Feature(UserInterface.cs): Добавлена сериализация и десериализация.

LarionovKirill committed:
Style(UserInterface.cs): незначительные изменения.

LarionovKirill committed:
Fix(UserInterface.cs): Испарвил баг с неправильным отображением времени и городов

LarionovKirill committed:
Feature(ChangingForm.cs): Изменений много. Я создал дополнительную форму для того, чтобы через нее добавлять и изменять значения, а главная только для обзора данных.

LarionovKirill committed:
Fix(MainForm.cs): исправлено название лейбла.
Fix(UserInterface.cs): Исправлен баг с вызовом одного и того же участка кода.
Feature(UserInterface.cs): добалена обработка исключений.

LarionovKirill committed:
Я случайно создал папку в папке и было пустое вложение

LarionovKirill committed:
Fix(UserInterface.cs): исправил выравнивание элементов.
Fix(MainForm.cs): изменил максимальный и минимальный размер формы.

LarionovKirill committed:
Refactoring(PersonalTask.sln): расположил формы в папки

LarionovKirill committed:
Feature(project): добавил папку и первую лабу по ООП.

LarionovKirill committed:
Я тупанул, но вроде все нормально вернул.

Ershovoy commented on PersonalTask/PersonalTask/View/Interfaces/UserInterface.cs:
```diff
@@ -0,0 +1,307 @@
+        {
+
+            var path = "AirTravels.json";
+            var jsonString = File.ReadAllText(path);
```
![image](https://user-images.githubusercontent.com/50287099/234297884-c48affc1-7ffa-4302-88da-70b04bb62fa2.png)

LarionovKirill replied:
Я забыл, что он в bin файле. Сейчас буду пробовать достать файл из ресурсов. Но не уверен, что получится.

Ershovoy commented on PersonalTask/PersonalTask/Model/AirTravel.cs:
```diff
@@ -0,0 +1,168 @@
+                PersonalTask.Services.ColissionManager.AssertDepartureTime(
+                    value,
+                    "Указанная дата не может быть установлена."
+                    );
```
Закрывающая скоба в параметрах фунции не переносится на отдельную строку, оставляй на предыдущей.

Ershovoy commented on PersonalTask/PersonalTask/Model/AirTravel.cs:
```diff
@@ -0,0 +1,168 @@
+            }
+            set
+            {
+                PersonalTask.Services.ColissionManager.AssertOnLengthOfString(
```
Можно сократить с PersonalTask.Services... до Services...
PersonalTask пространство имён относится ко всему проекту, так что упоминать его смысла внутри проекта явно нет. Читаемости кода это не добавляет, но из-за этого приходится больше печатать, а это оу шит как плохо. Я знаю тебе очень хочется, у всех же свой стиль кода йоу. Но не вини меня, вини общество.

LarionovKirill replied:
Бро, понимаю тебя, народ прогнил, во всём виноваты люди

Ershovoy replied:
Не забый про nameof(). Он поможет избавиться от константной строки, что в свою очередь обеспечит полностью автоматическое переименование поля.

Ershovoy commented on PersonalTask/PersonalTask/Model/FlightType.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public enum FlightType
+    { 
+        Внутренний,
+        Международный
```
Русский... Серьёзно?

LarionovKirill replied:
А что нет?

Ershovoy commented on PersonalTask/PersonalTask/Services/ColissionManager.cs:
```diff
@@ -0,0 +1,70 @@
+        /// <param name="text">Вводимый на проверку текст.</param>
+        /// <param name="maxLength">Максимальная длина строки.</param>
+        /// <param name="propertyName">Текст ошибки.</param>
+        public static void AssertOnLengthOfString(string text, int maxLength, string propertyName)
```
3 армугент называется propertyName, а в XML комметарии описано, что это текст ошибки.

Ershovoy commented on PersonalTask/PersonalTask/Model/AirTravel.cs:
```diff
@@ -0,0 +1,168 @@
+                PersonalTask.Services.ColissionManager.AssertOnLengthOfString(
+                    value,
+                    100,
+                    "Длина строки не должна превышать 100 символов.");
```
Текст ошибки лучше поместить внутрь метода, чтобы не приходилось на каждый вызов писать одно и тоже. Сделай сообщение обобщённым и передавай в аргументы имя свойства, чтобы программист знал какое свойство вызвало исключение.

Ershovoy replied:
Не забый про nameof(). Он поможет избавиться от константной строки, что в свою очередь обеспечит полностью автоматическое переименование поля.

LarionovKirill replied:
Ок, исправлю

Ershovoy commented on PersonalTask/PersonalTask/View/Interfaces/UserInterface.cs:
```diff
@@ -0,0 +1,307 @@
+        /// <summary>
+        /// Список авиаперелетов.
+        /// </summary>
+        List<Model.AirTravel> _airTravels = new List<Model.AirTravel>();
```
Вид доступа к члену класса всегда должен быть явно указан.

Ershovoy commented on PersonalTask/PersonalTask/View/Interfaces/UserInterface.cs:
```diff
@@ -0,0 +1,307 @@
+            {
+                try
+                {
+                    _airTravels.Add(new Model.AirTravel(
+                    addForm.departureTextBox.Text,
+                    addForm.destinationTextBox.Text,
+                    addForm.departureTime.Value,
+                    int.Parse(addForm.flightTimeTextBox.Text),
+                    (Model.FlightType)Enum.Parse(typeof(Model.FlightType),
+                    addForm.typeOfFlightComboBox.SelectedItem.ToString())));
+                    airTravelsListBox.Items.Add(CreateStringForList(_airTravels.Last()));
```
Ну что за позореще, мне. 
Каждая инструкция должна располагаться на отдельной строке, в таком случае код удобнее читать и пробегать по нему дебагером. Конечно не нужно прям каждую малешую инструкцию рассписывать на отдельные строки, но и в такой раш, как ты входить не нужно. Пять функций вызвал в одной строчке, пять, Карл!

Ershovoy commented on PersonalTask/PersonalTask/View/Interfaces/UserInterface.cs:
```diff
@@ -0,0 +1,307 @@
+            {
+                for (int j = 0; j < itearation - 1 - i; j++)
+                {
+                    if (_airTravels[j].DepartureTime > _airTravels[j + 1].DepartureTime)
```
У тебя же есть закрытое свойство для поля _airTravels, обращайся через него. Объяснял почему лучше так делать в прошлом или позапрошлом пуле.

LarionovKirill replied:
Тут я забыл про него видимо

Ershovoy commented on PersonalTask/PersonalTask/View/Interfaces/UserInterface.Designer.cs:
```diff
@@ -0,0 +1,246 @@
+        /// Требуемый метод для поддержки конструктора — не изменяйте 
+        /// содержимое этого метода с помощью редактора кода.
+        /// </summary>
+        private void InitializeComponent()
```
По заданию в программе должна быть адаптивная вёрстка.

LarionovKirill replied:
Я так и не разобрался со свойство anchor

Ershovoy commented on PersonalTask/PersonalTask/View/Interfaces/UserInterface.cs:
```diff
@@ -0,0 +1,307 @@
+        /// <summary>
+        /// Метод обрабатывает изменения выбранного перелета
+        /// </summary>
+        private void ChangePictureBox_Click(object sender, EventArgs e)
```
https://user-images.githubusercontent.com/50287099/234318039-b94903df-7bde-4a3c-b195-aaf810a089b6.mp4

LarionovKirill replied:
Исправил этот баг

LarionovKirill committed:
Fix(FlightType.cs): Изменил типы перелетов на английские.

LarionovKirill committed:
Fix(Project): Я без понятия что началось твориться на этом репозитории.

LarionovKirill committed:
Fix(UserInterface.cs): Исправил в методе обращение к свойству.

LarionovKirill committed:
Fix(CollisionManager.cs): Исправил XML комментарии и описание исключения.
Fix(AirTravel.cs): Исправил передачу аргументов в метод и убрал отступы закрывающей скобки.

LarionovKirill committed:
Fix(Project): Убрал PersonalTask из указания namespace.

LarionovKirill committed:
Fix(FlightType.cs): Изменил типы перелета на английские названия.

LarionovKirill committed:
Fix(UserInterface.cs): Исправил обращение к json файлу.

LarionovKirill committed:
Fix(UserInterface.cs): Исправил возможность вылета при добавлении перелета.

LarionovKirill committed:
Fix(UserInterface.cs): Исправил свою конструкцию в конструкторе на более мелкие и разбитые действия.

LarionovKirill committed:
Refactoring(Project): Малые изменения.

LarionovKirill committed:
.

LarionovKirill committed:
Fix(AirTravels.cs): Исправлена передача имени свойства в метод.
Fix(UserInterface.cs): Исправлено обращение к файлу. Теперь он хранится также в bin файле и создается при необходимости.
